### PR TITLE
tec: Add e2e tests (cypress)

### PR DESCRIPTION
**Pull request by marien-probesys**
Changes proposed in this pull request:

- Add and configure Cypress
- Add two commands to the Makefile (`make e2e-open` and `make e2e-run`)
- Configure the CI to execute the Cypress tests
    - it basically installs backend and frontend side-by-side, with Nginx as web server
    - then, it run the Cypress GitHub action on the frontend
    - the old `ci.yml` has been renamed `tests.yml`
- Add tests for login, organizations (list, creation and deletion) and users (list, creation and deletion)
- Remove the old Angular tests which are no longer useful (e2e tests should catch the same issues… and more)
- Unit test the OrganizationsSorter (the CI was failing because there were no tests any more to execute and the OrganizationsSorter is complex enough to be tested)

How to test the feature manually:

1. Update your dependencies with `make install`
2. Open Cypress with `make e2e-open` or run the tests with `make e2e-run`
3. Check that everything works fine

More information on how to configure e2e tests at: https://github.com/fusionSuite/documentation/pull/28

Pull request checklist:

- [x] code is manually tested
- [x] tests are updated
- [x] commit messages are relevant
- [x] documentation is updated (including code comments, commit messages…)

_If you think one of the item isn’t applicable to the PR, please check it
anyway and precise `N/A` next to it._

Related to https://github.com/fusionSuite/documentation/pull/28

### CODE REVIEW

**Comment by marien-probesys** [comment]
TODO checklist:

- [x] ~~allow to execute Cypress in Docker environment?~~
    - it wasn't desirable because it would have raised the Docker image from 160MB to 1,17GB
    - it wasn't even possible since Cypress has to run `reset` command on the backend side (in Docker, it would have been containerized)
- [x] allow to execute via the Makefile
- [x] allow to configure Cypress baseUrl (possible by default https://docs.cypress.io/guides/references/configuration#Overriding-Options)
- [x] configure the CI
    - [x] Install the backend & start it
    - [x] Build the frontend & start it
    - [x] Execute Cypress https://github.com/cypress-io/github-action
- [x] setup database reset & seed https://github.com/fusionSuite/backend/pull/27
    - [x] refactor
- [x] setup logging (see https://github.com/cypress-io/cypress-example-recipes#logging-in-recipes)
    - [x] refactor
- [x] write tests for organizations
- [x] write tests for users
- [x] ~~write components tests if necessary~~ I don't think it's necessary for the moment
- [x] remove Angular tests
- [x] write the documentation https://github.com/fusionSuite/documentation/pull/28

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+          MARIADB_USER: fusionsuite
+          MARIADB_PASSWORD: fusionsuite
+          MARIADB_ROOT_PASSWORD: fusionsuite
+
+    steps:
+    # Install PHP and NodeJs
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: '7.4'
```
Comment (ddurieux) [inline .github/workflows/cypress.yml:29]: php 7.4 is very near end of life. It could be better to use 8.0 or 8.1 directly

**Review by ddurieux — CHANGES_REQUESTED** [review]

File: .github/workflows/cypress.yml (hunk 1 above)
Comment (marien-probesys) [reply .github/workflows/cypress.yml:29]: I usually prefer to test with the oldest supported version (which is 7.4 according to the documentation :))

Comment (ddurieux) [reply .github/workflows/cypress.yml:29]: ok ;)

**Review by ddurieux — APPROVED** [review]

**Pull request merged**